[BUG] Can't initialize `AudioBundle` without `DJSystem`

## Description

After merging in my PR to separate out initialization of the `DJSystem`, I can't initialize just an `AudioBundle`.

## Reproduction Steps

Initialize an `AudioBundle` as follows:

```
    let game_data = GameDataBuilder::default()
        .with_bundle(RenderBundle::new(pipe, Some(config)))?
        .with_bundle(AudioBundle::new())?
        .with_bundle(TransformBundle::new())?
        .with_bundle(
            InputBundle::<String, String>::new().with_bindings_from_file(&key_bindings_path)?,
        )?
        .with_bundle(GameBundle)?;
```

## What You Expected to Happen

The `AudioBundle` initializes without a `DJSystem`.

## What Actually Happened

```
error[E0282]: type annotations needed                                                                                                                                                    
   --> src/main.rs:245:22                                                                                                                                                                
    |                                                                                                                                                                                    
245 |         .with_bundle(AudioBundle::new())?                                                                                                                                          
    |                      ^^^^^^^^^^^^^^^^ cannot infer type for `F`                                                                                                                    
                                                                                                                                                                                         
```


## Your Environment

1. Operating System: Linux
2. Operating System Version: 4.14
3. Version of Amethyst: master

## Additional Context

This is my PR, so happy to fix this. I just don't know how, nor would I like to provide a type annotation if I can avoid it. The tutorials/examples work, so things work if I use `DJSystem`, but the point of my fix was to *not* always have to use it. :) Thanks for any suggestions on a fix.

--- a/amethyst_audio/src/bundle.rs
+++ b/amethyst_audio/src/bundle.rs
@@ -1,71 +1,24 @@
 //! ECS audio bundles
 
-use std::marker::PhantomData;
-
-use rodio::default_output_device;
-
 use amethyst_assets::Processor;
 use amethyst_core::{
     bundle::{Result, SystemBundle},
     specs::prelude::DispatcherBuilder,
 };
 
-use crate::{source::*, systems::DjSystem};
+use crate::source::*;
 
 /// Audio bundle
 ///
-/// Will only register the `AudioSink` and the `DjSystem` if an audio output is found.
-/// `DjSystem` will be registered with name "dj_system".
+/// This will only add the asset processor for `Source`.
 ///
-/// This will also add the asset processor for `Source`.
+/// `DjSystem` must be added separately if you want to use our background music system.
 ///
-/// ## Errors
-///
-/// No errors returned by this bundle
-///
-/// ## Panics
-///
-/// Panics during `DjSystem` registration if the bundle is applied twice.
-///
-pub struct AudioBundle<'a, F, R> {
-    dep: &'a [&'a str],
-    marker: PhantomData<R>,
-    picker: Option<F>,
-}
+pub struct AudioBundle;
 
-impl<'a, F, R> AudioBundle<'a, F, R> {
-    /// Create a new audio bundle
-    pub fn new() -> Self {
-        AudioBundle {
-            dep: &[],
-            marker: PhantomData,
-            picker: None,
-        }
-    }
-
-    /// Initialize the DJ system
-    pub fn with_dj_system(mut self, picker: F) -> Self {
-        self.picker = Some(picker);
-        self
-    }
-
-    /// Set dependencies for the `DjSystem`
-    pub fn with_dep(mut self, dep: &'a [&'a str]) -> Self {
-        self.dep = dep;
-        self
-    }
-}
-
-impl<'a, 'b, 'c, F, R> SystemBundle<'a, 'b> for AudioBundle<'c, F, R>
-where
-    F: FnMut(&mut R) -> Option<SourceHandle> + Send + 'static,
-    R: Send + Sync + 'static,
-{
+impl<'a, 'b> SystemBundle<'a, 'b> for AudioBundle {
     fn build(self, builder: &mut DispatcherBuilder<'a, 'b>) -> Result<()> {
         builder.add(Processor::<Source>::new(), "source_processor", &[]);
-        if default_output_device().is_some() && self.picker.is_some() {
-            builder.add(DjSystem::new(self.picker.unwrap()), "dj_system", self.dep);
-        }
         Ok(())
     }
 }

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -19,14 +19,14 @@ The format is based on [Keep a Changelog][kc], and this project adheres to
 * `amethyst_renderer::Rgba` is now a `Component` that changes the color and transparency of the entity
 it is attached to. ([#1282])
 * `AutoFov` and `AutoFovSystem` to adjust horizontal FOV to screen aspect ratio. ([#1281])
-* `AudioBundle.with_dj_system(...)` initializes the `DjSystem` as part of the `AudioBundle`. ([#1346])
+
 
 ### Changed
 
 * Make `application_root_dir` return a `Result<Path>` instead of a `String` ([#1213])
 * Remove unnecessary texture coordinates offset in `Sprite::from_pixel_values` ([#1267])
 * Changed `ActiveCamera` to have the `Option` inside. ([#1280])
-* `AudioBundle::new()` accepts no arguments and no longer initializes a `DjSystem`. ([#1346])
+* `AudioBundle::new()` no longer exists, as `AudioBundle` is now a unit type. It also no longer initializes the `DjSystem` ([#1356])
 
 ### Removed
 
@@ -46,7 +46,7 @@ it is attached to. ([#1282])
 [#1281]: https://github.com/amethyst/amethyst/pull/1281
 [#1302]: https://github.com/amethyst/amethyst/pull/1302
 [#1328]: https://github.com/amethyst/amethyst/pull/1328
-[#1346]: https://github.com/amethyst/amethyst/pull/1346
+[#1356]: https://github.com/amethyst/amethyst/pull/1356
 
 ## [0.10.0] - 2018-12
 

--- a/examples/appendix_a/main.rs
+++ b/examples/appendix_a/main.rs
@@ -12,7 +12,7 @@ mod systems;
 
 use crate::{audio::Music, bundle::PongBundle, config::PongConfig};
 use amethyst::{
-    audio::AudioBundle,
+    audio::{AudioBundle, DjSystem},
     core::{frame_limiter::FrameRateLimitStrategy, transform::TransformBundle},
     ecs::prelude::{Component, DenseVecStorage},
     input::InputBundle,
@@ -51,7 +51,12 @@ fn main() -> amethyst::Result<()> {
         )?
         .with_bundle(PongBundle::default())?
         .with_bundle(TransformBundle::new().with_dep(&["ball_system", "paddle_system"]))?
-        .with_bundle(AudioBundle::new().with_dj_system(|music: &mut Music| music.music.next()))?
+        .with_bundle(AudioBundle)?
+        .with(
+            DjSystem::new(|music: &mut Music| music.music.next()),
+            "dj_system",
+            &[],
+        )
         .with_bundle(UiBundle::<String, String>::new())?
         .with_basic_renderer(display_config_path, DrawFlat::<PosTex>::new(), true)?;
 

--- a/examples/pong/main.rs
+++ b/examples/pong/main.rs
@@ -8,7 +8,7 @@ mod pong;
 mod systems;
 
 use amethyst::{
-    audio::AudioBundle,
+    audio::{AudioBundle, DjSystem},
     core::{frame_limiter::FrameRateLimitStrategy, transform::TransformBundle},
     ecs::prelude::{Component, DenseVecStorage},
     input::InputBundle,
@@ -72,7 +72,12 @@ fn main() -> amethyst::Result<()> {
         .with_bundle(PongBundle)?
         .with_bundle(RenderBundle::new(pipe, Some(config)).with_sprite_sheet_processor())?
         .with_bundle(TransformBundle::new().with_dep(&["ball_system", "paddle_system"]))?
-        .with_bundle(AudioBundle::new().with_dj_system(|music: &mut Music| music.music.next()))?
+        .with_bundle(AudioBundle)?
+        .with(
+            DjSystem::new(|music: &mut Music| music.music.next()),
+            "dj_system",
+            &[],
+        )
         .with_bundle(UiBundle::<String, String>::new())?;
     let mut game = Application::build(assets_dir, Pong)?
         .with_frame_limit(

